backend/routers/chat.py

```
@@ -0,0 +1,181 @@
+    try:
+        service = ChatService(redis_client)
+        remaining, limit = await service.check_rate_limit(identity)
+    except RateLimitExceeded as exc:
+        tier_label = "訪客" if identity.tier == "guest" else "用戶"
+        raise HTTPException(
+            status_code=429,
+            detail={
+                "detail": f"每日問答次數已達上限（{tier_label}：{exc.limit}次/天）",
+                "limit": exc.limit,
+            },
+        )
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)